src/main/shell-session.ts

```diff
@@ -65,11 +62,13 @@ export class ShellSession extends EventEmitter {
   }
 
   protected cwd(): string {
-    if(!this.preferences || !this.preferences.terminalCWD || this.preferences.terminalCWD === "") {
+    const { preferences } = this.cluster;
+
+    if(!preferences || !preferences.terminalCWD || preferences.terminalCWD === "") {
       return null;
     }
 
-    return this.preferences.terminalCWD;
+    return preferences.terminalCWD;
```

Comment:
how about:

```js
    return preferences?.terminalCWD || null;
```

Reply:
I did only minimal change (`this.preferences` -> `this.cluster.preferences`) hesitated to do any additional refactorings.